public/app/features/api-keys/state/actions.ts

```diff
@@ -32,10 +32,19 @@ export function migrateApiKey(id: number): ThunkResult<void> {
   };
 }
 
-export function migrateAll(): ThunkResult<void> {
+interface MigrationResult {
+  Total: number;
+  Migrated: number;
+  Failed: number;
+  FailedApikeyIDs: number[];
+  FailedDetails: string[];
+}
+
+export function migrateAll(): ThunkResult<Promise<MigrationResult>> {
   return async (dispatch) => {
     try {
-      await getBackendSrv().post('/api/serviceaccounts/migrate');
+      const request = await getBackendSrv().post('/api/serviceaccounts/migrate');
+      return request;
     } finally {
```

Comment:
Why finally is here, but without catch?

Reply:
it's without a catch currently - https://github.com/grafana/grafana/blob/main/public/app/features/api-keys/state/actions.ts#L38

Reply:
this was because we didnt want to handle errors initially. this request gets back a result so no errors are thrown and caught still